simulator/controller/controller.go

```diff
@@ -0,0 +1,154 @@
+package controller
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"golang.org/x/xerrors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/metadata"
+	"k8s.io/client-go/metadata/metadatainformer"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/controller-manager/pkg/clientbuilder"
+	"k8s.io/controller-manager/pkg/informerfactory"
+	"k8s.io/klog/v2"
+	controllermanageroptions "k8s.io/kubernetes/cmd/kube-controller-manager/app/options"
+	kubectrlmgrconfig "k8s.io/kubernetes/pkg/controller/apis/config"
+)
+
+const (
+	// controllerStartJitter is the Jitter used when starting controller.
+	controllerStartJitter = 1.0
+)
+
+type initFunc func(ctx context.Context, controllerCtx controllerContext) error
+
+// controllerInitializersFunc is used to create a collection of initializers
+// given the loopMode.
+
+type controllerInitializersFunc func() (initializers map[string]initFunc)
+
+var _ controllerInitializersFunc = newControllerInitializers
```

Comment:
Why do we. need this `controllerInitializersFunc` type?

Reply:
No. It's my mistake.
I forgot to remove it when simplifying the code I used as an example. Currently, we won't need it.